src/NHibernate/Hql/Ast/ANTLR/QueryTranslatorImpl.cs

```diff
@@ -283,9 +284,29 @@ public bool ContainsCollectionFetches
 		{
 			get
 			{
-				ErrorIfDML();
-				IList<IASTNode> collectionFetches = ((QueryNode)_sqlAst).FromClause.GetCollectionFetches();
-				return collectionFetches != null && collectionFetches.Count > 0;
+				return CollectionFetches.Count > 0;
+			}
+		}
+
+		public bool CanAddFetchedCollectionToCache
+		{
+			get
+			{
+				foreach (var collectionFetch in CollectionFetches)
```

Comment:
Before iterating the collections, it would be nice if the overall cacheability of the query was checked first. (But maybe that check should be done in the loader.)

Reply:
Checking the global cacheability of the query does not seem doable where this property is used, because it requires the queryParameters, which are not available there.